src/modules/ekf2/EKF2Selector.cpp

```diff
@@ -733,7 +734,9 @@ void EKF2Selector::Run()
 				SelectInstance(best_ekf);
 			}
 
-		} else if (lower_error_available && (hrt_elapsed_time(&_last_instance_change) > 10_s)) {
+		} else if (lower_error_available
+			   && ((hrt_elapsed_time(&_last_instance_change) > 10_s) || _instance[_selected_instance].warning)) {
```

Comment:
I don't think we want to switch if we get a single observation that exceeds the innovation test threshold. We can afford to wait a second because the EKF will ignore that observation and inertial errors take a few seconds to build.